### ISSUE TITLE
bug: react swipe back in tab goes to previous tab

### ISSUE THREAD

**Issue by rgoldiez** [ionic-team/ionic-framework#21993]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

We've run into an issue that we can reproduce on the conference app example.  It looks like on the *latest* version of Ionic (5.3.1) when you swipe to open the menu it navigates back.  Note that the stock conference app is on an older version (5.0.x) of the framework that does *not* exhibit this issue.  You'll need to upgrade the framework in the conference app to 

**Expected behavior:**

Swiping to open the menu should not alter location history.

**Steps to reproduce:**

Noted the animated GIF below.  I added logging to watch `location.pathname` in the menu.  You can see it changes when trying to swipe to open the menu.
![Aug-27-2020 11-42-00](https://user-images.githubusercontent.com/10797610/91470520-6c4e6880-e85a-11ea-83c7-437c52e5e0f2.gif)

**Related code:**

The example conference app can be used without modification, but with the latest Ionic framework (5.3.1) to reproduce.

**Other information:**

n/a

**Ionic info:**

```
Ionic:

   Ionic CLI       : 6.11.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 5.3.1

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 2.4.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v12.16.2 (/usr/local/bin/node)
   npm    : 6.14.4
   OS     : macOS Catalina
```

**Comment by cmer4** [comment]
This was unfixable with iOS 13 release as the gestures were system level when introduced;/
But seems like with iOS 13.4 it is not possible to preventDefault behavior: https://pqina.nl/blog/blocking-navigation-gestures-on-ios-13-4/, did not check that though thoroughly.

**Comment by rgoldiez** [comment]
I'm a little confused by your comment @cmer4.  On iOS 13.4, the Ionic conference app (the example app) works fine with it's default version of ionic (`5.0.5`).  But when upgrading to the latest version of Ionic (`5.3.1`, still on iOS 13.4) the gesture to open the menu navigates back.  Also note that this is running in capacitor on iOS.

**Comment by cmer4** [comment]
@rgoldiez hmm this is confusing indeed. I just tested my older version of the app and in my case the navigation gesture always works if event.preventDefault() is not called.

**Comment by rgoldiez** [comment]
Bumping this.  Any plans or ETA to address this @liamdebeasi ?

**Comment by yyesi** [comment]
@rgoldiez the reason for this behavior, that under the hood on tab click `history.push()` is being used instead of `history.replace()` as a workaround you can use a little dirty hack by overriding `handleIonTabButtonClick` function that is being used as `onClick` handler on  `<IonTabButton/>`

```
 const handleTabClick: any = (href: string) => (ref: any) => {
      if (!ref) {
        return;
      }
      ref.handleIonTabButtonClick = () => {
        history.replace(href);
      };
  }

 <IonTabButton ref={handleTabClick('/some-route-link')}
</IonTabButton>
```

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue was fixed in Ionic 6.1.15, but this thread was missed when closing relevant issues out. I recommend updating to the latest version of Ionic 6 to receive this fix. Let me know if there are any questions. Thanks!
